src/lib/spacefinder/liveblog-right-column-adverts.ts

```diff
@@ -55,20 +53,54 @@ const fillAdvertSlots = async (numAdsToInsert: number, fromIndex: number) => {
 	}
 };
 
-// When a liveblog-right ad slot is inserted into the page, create an ad in this new slot
-export const initLiveblogRightAdverts = (): Promise<void> => {
-	if (isInVariantSynchronous(liveblogRightColumnAds, 'multiple-adverts')) {
-		return Promise.resolve();
-	}
+const createLiveblogRightAdverts = () => {
+	void fillAdvertSlots();
 
 	document.addEventListener('liveblog-right-ads-inserted', (event: Event) => {
 		if (!isCustomEvent(event)) throw new Error('not a custom event');
 
 		const { numAdsToInsert, fromIndex } = (<AdsInsertedCustomEvent>event)
 			.detail;
 
-		void fillAdvertSlots(numAdsToInsert, fromIndex);
+		// Create an array with the indexes of adverts that need to be created
+		// This is to exclude adverts that have already been loaded
+		const newAdvertIndexes = [...Array(numAdsToInsert).keys()].map(
+			(i) => i + fromIndex,
+		);
+
+		void fillAdvertSlots(newAdvertIndexes);
 	});
+};
+
+const restrictRightAdContainerHeight = () => {
+	void fastdom.measure(() => {
+		const rightAdContainer =
+			document.querySelector<HTMLElement>('#top-right-ad-slot');
+		if (rightAdContainer !== null) {
+			rightAdContainer.style.maxHeight = '1059px';
+		}
```

Comment:
Does this part need to be in a `fastdom.mutate` call rather than measure?

Reply:
Yes, good spot